### PR TITLE
feat(docs): add gitbook

### PR DESCRIPTION
**Pull request by Jimexist**
**What this PR does**
add skeleton for docs and website

**why we need it**
for PR of course :-)

**Special notes for your reviewer**:
pertaining to #39

### CODE REVIEW

File: docs/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -euf -o pipefail
+gitbook build
+rm -rf ../docs
```
Comment (ddysher) [inline docs/build.sh]: is it safe to always remove the directory?

File: website/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -euf -o pipefail
+gitbook build
+rm -rf ../docs
```
Comment (ddysher) [inline website/build.sh]: same here?

File: docs/build.sh (hunk 1 above)
Comment (Jimexist) [reply docs/build.sh]: it is... because `/docs` is auto-generated and there's git to track history.

File: website/build.sh (hunk 2 above)
Comment (Jimexist) [reply website/build.sh]: it is... because `/docs` is auto-generated and there's git to track history.

**Comment by Jimexist** [comment]
@ddysher 
> Are we going to put all docs under website?

`.md`s under `webiste`, generated gitbook under `/docs` so that GitHub pages may pickup

File: docs/build.sh (hunk 1 above)
Comment (ddysher) [reply docs/build.sh]: yeah, it's auto-generated here, but it is quite a established rules to put docs under `docs` directory.

**Pull request merged**